backend-api/src/functions/asyncBiometricToken/asyncBiometricTokenHandler.test.ts

```diff
@@ -1,23 +1,71 @@
+import { Context } from "aws-lambda";
 import { Logger } from "../services/logging/logger";
 import { MockLoggingAdapter } from "../services/logging/tests/mockLogger";
 import { buildLambdaContext } from "../testUtils/mockContext";
 import { buildRequest } from "../testUtils/mockRequest";
 import { lambdaHandlerConstructor } from "./asyncBiometricTokenHandler";
-import { registeredLogs } from "./registeredLogs";
+import { IAsyncBiometricTokenDependencies } from "./handlerDependencies";
+import { MessageName, registeredLogs } from "./registeredLogs";
 
 describe("Async Biometric Token", () => {
-  describe("Given a request is made", () => {
+  let dependencies: IAsyncBiometricTokenDependencies;
+  let mockLoggingAdapter: MockLoggingAdapter<MessageName>;
+  let context: Context;
+
+  beforeEach(() => {
+    mockLoggingAdapter = new MockLoggingAdapter();
+    dependencies = {
+      logger: () => new Logger(mockLoggingAdapter, registeredLogs),
+    };
+    context = buildLambdaContext();
+  });
+  describe("Request body validation", () => {
+    describe("Given request body is invalid", () => {
+      it("Logs and returns 400 Bad Request response", async () => {
```

Comment:
New idea (not a blocker for approval or anything): ideally, I think unit tests should aim to test just one thing at a time (and therefore to fail for just one reason), which makes catching bugs more straightforward among other things. This test checks both the logs and the lambda result. We could instead use a beforeEach or beforeAll block to invoke the function under test (and capture its result), with individual it blocks to assert on different things we care about (logs vs return value).

Reply:
Done

Reply:
Have a look and see what you think now.

We will be doubling the amount of times we execute the functions this was using a beforeEach, is that a problem? 

I did try to use beforeAll, but that messed with the beforeEach in the top level block. I think the beforeAll in this block ran before the beforeEach in the top level which mean that the test failed due to things to being created in that top level beforeEach.

Feel free to improve, leave it as it is, or revert this change if you want (`c7321c704d636fa874c486923c7f0137371aa80b`)

Reply:
Kept the testing approach just did away with the beforeEach.